p4_symbolic/ir/test.bzl

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains rule definitions for running
+# the test binary to produce output json and protobuf files,
+# subset diff the input and output json files, and golden file
+# testing the output protobuf files against the expected files.
+# It also defines a Macro that simplifies defining a protobuf test
+# over a sample P4 program.
+
+load("//:p4c.bzl", "run_p4c")
+load("//p4_symbolic/bmv2:test.bzl", "exact_diff_test")
+
+# Macro that defines exact diff IR testing rules for a given p4 program
+# with all their dependent rules.
+# The macro defines these rules in order:
+# 1. A rule for producing bmv2 json and p4info files from a .p4 file using p4c.
+# 2. A rule for parsing the bmv2 json and p4info using p4_symbolic/main.cc, and
+#    dumping a protobuf output file.
+# 3. A rule for golden file testing of the protobuf output file against
+#    the specified expected file.
+# Use the p4_deps list to specify dependent files that p4_program input
+# file depends on (e.g. by including them).
+def ir_parsing_test(name, p4_program, golden_file, p4_deps = []):
```

Comment:
I am not very opposed to this, but if we keep it this, it will be very similar to the tests under bmv2/

The bmv2/ tests need to be this way otherwise they will look too ugly and have too many duplicates.

I think there is value in keeping these tests similar to the bmv2 ones.

Reply:
Let us revisit this when we revisit the other tests here (the ones that are currently errors and not protobufs)
I am honestly not sure about this one way or another.